src/stdune/path.ml

```diff
@@ -1163,22 +1163,26 @@ let insert_after_build_dir_exn =
     | External _ ->
       error a b
 
-let rm_rf =
-  let rec loop dir =
-    Array.iter (Sys.readdir dir) ~f:(fun fn ->
-        let fn = Filename.concat dir fn in
-        match Unix.lstat fn with
-        | { st_kind = S_DIR; _ } -> loop fn
-        | _ -> unlink_operation fn);
-    Unix.rmdir dir
-  in
-  fun ?(allow_external = false) t ->
-    if (not allow_external) && not (is_managed t) then
-      Code_error.raise "Path.rm_rf called on external dir" [ ("t", to_dyn t) ];
-    let fn = to_string t in
-    match Unix.lstat fn with
-    | exception Unix.Unix_error (ENOENT, _, _) -> ()
-    | _ -> loop fn
+let rec clear dir =
```

Comment:
Would it be better to rename to `clear_dir` or `clear ~dir` so it's clear that clearing a file is not meaningful?

By the way, I can now see that `rm_rf` will crash when applied to a file, maybe worth renaming that too or fixing this pitfall. (I do realize it's not a new issue)

Reply:
I renamed `clear` to `clear_dir`. I'm also considering adding a `Dir` submodule to `Path` to collect these dir only functions.

I fixed rm_rf as well.